src/openrct2/interface/Window.cpp

```diff
@@ -155,6 +155,43 @@ static constexpr float kWindowScrollLocations[][2] = {
 
         auto windowManager = Ui::GetWindowManager();
         windowManager->UpdateMouseWheel();
+
+        // Update window and viewport visibilities
+        {
+            const auto itEnd = g_window_list.end();
+            for (auto it = g_window_list.begin(); it != itEnd; ++it)
+            {
+                auto& window = *(*it);
+                if (window.viewport == nullptr)
+                {
+                    window.isVisible = true;
+                    continue;
+                }
+                if (window.classification == WindowClass::MainWindow)
+                {
+                    window.isVisible = true;
+                    window.viewport->isVisible = true;
+                    continue;
+                }
+                window.isVisible = true;
+                window.viewport->isVisible = true;
+                for (auto itOther = std::next(it); itOther != itEnd; ++itOther)
+                {
+                    const auto& otherWindow = *(*itOther);
+                    if (otherWindow.flags & WF_DEAD)
+                        continue;
+
+                    if (otherWindow.windowPos.x <= window.windowPos.x && otherWindow.windowPos.y <= window.windowPos.y
+                        && otherWindow.windowPos.x + otherWindow.width >= window.windowPos.x + window.width
+                        && otherWindow.windowPos.y + otherWindow.height >= window.windowPos.y + window.height)
+                    {
+                        window.isVisible = false;
+                        window.viewport->isVisible = false;
+                        break;
+                    }
+                }
+            }
+        }
```

Comment:
Quite a bit of code, perhaps its best moved into a new function, something like `WindowsUpdateVisibilities`